### PR TITLE
Fix non required json input being set to 'null' instead of null

### PR DESCRIPTION
**Pull request by markkaylor**
### What does it do?

- Fixes null being set as a string when it should be set as `null`

### Why is it needed?

- It creates a bad editing experience

### How to test it?
- Run yarn setup
- JSONInput should be empty by default
- A non-required json input should be empty by default (not `null`)
- A non-required json input can be left empty but will be set with null on save
- Emptying the json input will not fill the input with "null", but it will instead remain empty and filled with null on save and refetch.
- A required json field should be empty by default (not `{}`)
- A required json field cannot be left empty and will throw a validation error: "value is required"

### Related issue(s)/PR(s)
resolves: https://github.com/strapi/strapi/issues/15776
resolves: https://github.com/strapi/strapi/issues/15207

### CODE REVIEW

**Comment by gu-stav** [comment]
Would this fix https://github.com/strapi/strapi/issues/15207 too?

**Comment by markkaylor** [comment]
@gustav at a quick glance I think it would but I'll test in a bit to make sure

**Review by gu-stav — COMMENTED** [review]
Something is not right here.

1) When I create a new entity which has two JSON fields (one required and one not required): the non-required has a default value of `null` and the required of `{}`. Both fields should imho be empty by default.

2) When I edit an entity and delete `null` from the non-required field when editing an entity the Save button is not enabled and therefore I can not save.

3) when I empty a JSON field, when had a value previously it still prints `null`.

**Comment by markkaylor** [comment]
Sorry Gustav I forgot to mention in the how to test to run `yarn setup` since the change is in the helper plugin, I'll update the PR description. Or at least I don't think I have these two issues you are describing:

> 2. When I edit an entity and delete null from the non-required field when editing an entity the Save button is not enabled and therefore I can not save.
> 
> 3. when I empty a JSON field, when had a value previously it still prints null.

I recorded a video for 4.5.6 and this branch each with the keyboard input displayed:

4.5.6:

https://user-images.githubusercontent.com/26598053/218099558-006be798-1320-46c2-8b4f-2678804d3bba.mov

This branch:

https://user-images.githubusercontent.com/26598053/218099597-f69d5975-a5ff-40ac-a334-d1a49a2536f6.mov

For

> 1. When I create a new entity which has two JSON fields (one required and one not required): the non-required has a default value of null and the required of {}. Both fields should imho be empty by default

I agree that is a bit strange and I will open a PR in the DS so we don't set a default value of {}

**Comment by markkaylor** [comment]
@gu-stav 

I found where the default values were being set in the CM and removed them:

- The JSONInput now defaults to empty for both required and non-required
- Non-required will be set to null on save if no input was provided
- Required will throw a validation error if no input was provided

I confirm this PR also fixes https://github.com/strapi/strapi/issues/15207
- The new locale will be empty by default
- The new locale can be filled from another locale

**Review by gu-stav — COMMENTED** [review]
Even after rebuilding the helper-plugin I see `null` appearing when I'm saving the entity?

https://user-images.githubusercontent.com/2244375/218999209-cce6b3e2-0676-42fa-923e-6c3ab609d186.mp4

The default value case looks better now - thanks!

**Comment by markkaylor** [comment]
> Even after rebuilding the helper-plugin I see null appearing when I'm saving the entity?

oh ok I understand better now, thanks. I thought you were getting null onChange when you deleted, not when saving.

I thought this was the expected, or at least an acceptable, behavior. We have to save something for a non-required input and to me null makes more sense than an empty string. If it was an empty string you will see `""` there.

File: packages/core/helper-plugin/lib/src/components/GenericInput/index.js [hunk 1]
```diff
@@ -141,7 +141,7 @@ const GenericInput = ({
         <JSONInput
           label={label}
           labelAction={labelAction}
-          value={value}
+          value={value === 'null' ? null : value}
```
Comment (gu-stav) [inline packages/core/helper-plugin/lib/src/components/GenericInput/index.js:144]: Where does that 'null' come from? Does the API send us that?

**Comment by markkaylor** [comment]
@gu-stav I was not able to find a way for the editor to be empty and not display null on refetch while also still allowing null as a value the user can add to the input.

Currently I believe the behavior is now the same as it was before in Strapi before the JSONInput component was added where the value is set to null when empty and then filled with null on refetch. 

Otherwise I think this PR resolves the two issues mentioned and we could merge. I'm a bit out of time since it's near the end of day for me and I'll be on vacation for two weeks but would be happy to continue looking into this to improve the experience when I get back.

**Comment by gu-stav** [comment]
@remidej See Marks last comment: I think there is still outstanding work but I won't be able to take it over.

**Comment by markkaylor** [comment]
@gu-stav this PR fixes the bug, I think what you are requesting is an enhancement. The behavior is now how it was before the the json component was added where null is set when the value is empty. I'm not sure I understand why we don't move forward with this PR to resolve the bug as quickly as possible for our users and handle the enhancement you are requesting in another PR. In my opinion it is less of a priority. Have users complained in the past about the fact that null is set automatically when the value is empty?

**Comment by gu-stav** [comment]
@markkaylor

1. It wasn't clear to me this PR is ready to be reviewed and merged. If I misunderstood your last comment, I apologize.
2. It is the responsibility of the PR creator to merge - I've been on vacation and deep into other things. If you want the PR to merge faster, please assign someone with the bandwidth and context to merge it.

I can try to review again next week - the PR is also open to everyone else, so I shouldn't be the blocking factor and I haven't requested changes.

**Comment by markkaylor** [comment]
@gu-stav no problem, I think the communication just broke down somewhere along the line, sorry about that. I will get a review from Rémi and Fernando to move this one forward and add your feedback to a new ticket in our backlog. Sorry again for any confusion.

**Review by remidej — APPROVED** [review]
I was also under the impression that Gustav's feedback was blocking, so I also misunderstood the situation, sorry. Looks ready to merge to me to fix the bug in the next release

**Review by oiorain — APPROVED** [review]
QA LGTM 👌

**Pull request merged**